webapp/client/src/pages/LoginPage.js

```diff
@@ -1,60 +1,94 @@
 import PropTypes from "prop-types";
 import React from "react";
+import { useTranslation } from "react-i18next";
+import FormFieldIdNr from "../components/FormFieldIdNr";
+import FormFieldUnlockCode from "../components/FormFieldUnlockCode";
 import FormHeader from "../components/FormHeader";
+import FormRowCentered from "../components/FormRowCentered";
 import StepForm from "../components/StepForm";
 import StepHeaderButtons from "../components/StepHeaderButtons";
 
-export default function LoginPage({
-  prevUrl,
-  backLinkText,
-  stepTitle,
-  stepIntro,
-  submitUrl,
-  overviewButton,
-  csrfToken,
-  explanatoryButtonText,
-  nextButtonLabel,
-}) {
+export default function LoginPage({ backLink, stepHeader, form, fields }) {
+  const { t } = useTranslation();
+
   return (
-    <React.Fragment>
-      <StepHeaderButtons backLinkUrl={prevUrl} backLinkText={backLinkText} />
-      <FormHeader title={stepTitle} intro={stepIntro} />
-      <StepForm
-        {...{
-          submitUrl,
-          overviewButton,
-          csrfToken,
-          explanatoryButtonText,
-          nextButtonLabel,
-        }}
-      />
-    </React.Fragment>
+    <>
+      <StepHeaderButtons {...backLink} />
+      <FormHeader {...stepHeader} />
+      <StepForm {...form}>
+        <FormRowCentered>
+          <FormFieldIdNr
+            autofocus
+            required
+            fieldName="idnr"
+            // TODO: is the fieldId ever different from the fieldName?
+            fieldId="idnr"
+            values={fields.idnr.value}
+            label={{
+              text: t("unlockCodeActivation.idnr.labelText"),
+            }}
+            details={{
+              title: t("unlockCodeActivation.idnr.help.title"),
+              text: t("unlockCodeActivation.idnr.help.text"),
+            }}
```

Comment:
So we now set this in the component, not in the field itself? Because we technically could do:
```suggestion
            label={{text: fields.idnr.label}}
            details={{
              title: fields.idnr.details.title,
              text: fields.idnr.details.text,
            }}
```
Right?
Same goes for the name and ID if I'm not mistaken?

Reply:
We don't have access to `fields` here! That's the whole point! :)
The Python could set these in the step, but I kind of think UI strings belong into the UI? What do you think?